include/oneapi/dpl/pstl/hetero/dpcpp/unseq_backend_sycl.h

```diff
@@ -957,22 +957,22 @@ struct __brick_shift_left
 
 struct __brick_assign_key_position
 {
-    // __a is a tuple {i, i-th+1 key, i-th key}
+    // __a is a tuple {i, (i-1)-th key, i-th key}
     // __b is a tuple {key, index} that stores the key and index where a new segment begins
     template <typename _T1, typename _T2>
     void
     operator()(const _T1& __a, _T2&& __b) const
     {
-        ::std::get<0>(::std::forward<_T2>(__b)) = ::std::get<2>(__a);     // store new key value
-        ::std::get<1>(::std::forward<_T2>(__b)) = ::std::get<0>(__a) + 1; // store index of new key
+        ::std::get<0>(::std::forward<_T2>(__b)) = ::std::get<2>(__a); // store new key value
+        ::std::get<1>(::std::forward<_T2>(__b)) = ::std::get<0>(__a); // store index of new key
```

Comment:
According to the change in the comment `i-th+1 key` => `(i-1)-th`.....
Is it correct that we just remove `+ 1` ? Whether should be `- 1`?
Or all is OK here?
(I didn not re-check  whole algorithm again in details, but is looks suspicions, like a typo...)